pkgs/stdenv/darwin/make-bootstrap-tools.nix

```diff
@@ -1,5 +1,8 @@
 { pkgspath ? ../../.., test-pkgspath ? pkgspath
 , localSystem ? { system = builtins.currentSystem; }
+# Specify the desired LLVM version in an overlay to avoid the use of
+# mismatching versions.
```

Comment:
Could you clarify here what is mismatched with what?

As I understand it, prior to this PR, there was only one version of `llvmPackages` visible in this derivation.  It might not have been `llvmPackages_11`, but at least there was only one version.

What is the source of the "11" constraint?

Reply:
The `llvmPackages` that we take things (clang, libc++ and such) from is specified explicitly to be `llvmPackages_11` to keep the bootstrap-tools stable.

However, tools like otool, install_name_tool and strip are taken straight from `stdenv.cc`, which, after the bump, is a different LLVM version altogether.

This is what caused the mismatch and the simplest solution seems to me to simply get everything using the same LLVM version. The original intent was that bootstrap-tools specified LLVM 11 exhaustively but it didn't. That should be rectified with this PR. As to why stick with 11? That's just to keep the bootstrap-tools unchanged. There's no benefit to changing the bootstrap-tools unless we upload the tarball and use it as the bootstrap for the Darwin stdenv and that causes the most massive of rebuilds. So until we have a good reason to change the bootstrap-tools, I'd rather keep it as is.

Reply:
Thanks for the great explanation, @toonn!

I mostly wanted to have something in the code to explain this, so I copied your explanation into a comment in 985ff1cb889d5a438c6812ed827f375c1ee92b99 and appended it to this PR.  Hope that is okay.